### PR TITLE
flip track widget creation on by default

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This should have been flipped a long time ago. Hasn't been noticed since most IDE's specify true by default

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Waiting to see what this breaks ....

File: packages/flutter_tools/test/commands.shard/permeable/test_test.dart [hunk 1]
```diff
@@ -62,7 +62,8 @@ void main() {
 
     testUsingContext('report which user created widget caused the error - no flag', () async {
       Cache.flutterRoot = '../..';
-      return _testFile('print_user_created_ancestor_no_flag', automatedTestsDirectory, flutterTestDirectory);
+      return _testFile('print_user_created_ancestor_no_flag', automatedTestsDirectory, flutterTestDirectory,
+         extraArguments: const <String>['--no-track-widget-creation']);
```
Comment (jacob314) [inline packages/flutter_tools/test/commands.shard/permeable/test_test.dart]: we should run tests with --track-widget-creation by default. It provides a better experience when your tests have rendering errors as you can figure out where in your source code you likely triggered the error.

**Review by jacob314 — APPROVED** [review]
lgtm
Lgtm with a nit that we should be turning this on for tests as well.

File: packages/flutter_tools/test/commands.shard/permeable/test_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/commands.shard/permeable/test_test.dart]: Tests use the same track-widget-creation setting so they are on by default too. This test case is just testing the stack traces for twc being off, which is why I needed to switch the flag

Comment (jacob314) [reply packages/flutter_tools/test/commands.shard/permeable/test_test.dart]: sgtm

**Pull request merged**